cc/crypto/encryption_key_provider.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2023 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CC_CRYPTO_ENCRYPTION_KEY_PROVIDER_H_
+#define CC_CRYPTO_ENCRYPTION_KEY_PROVIDER_H_
+
+#include <memory>
+#include <string>
+#include <tuple>
+
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+#include "cc/crypto/hpke/recipient_context.h"
+
+namespace oak::crypto {
+
+class RecipientContextGenerator {
+ public:
+  virtual absl::StatusOr<RecipientContext> GenerateRecipientContext(
+      absl::string_view serialized_encapsulated_public_key) = 0;
+};
+
+class EncryptionKeyProvider : public RecipientContextGenerator {
+ public:
+  static absl::StatusOr<std::shared_ptr<EncryptionKeyProvider>> Create();
```

Comment:
I'm not sure why we need this static method to create an EncryptionKeyProvider instance. Why not move the 'Create()' logic to the constructor?

Reply:
The `Create` function can return an error, because private key generation can fail

Reply:
Got it, thanks.